providers/nitro/nitro-helper/src/command/nitro_enclave.rs

```diff
@@ -0,0 +1,164 @@
+        ));
+    }
+    serde_json::from_slice(output.stdout.as_slice())
+        .map_err(|_| "command invalid output".to_string())
+}
+
+fn run_enclave_daemon(
+    image_path: &str,
+    cpu_count: usize,
+    memory_mib: u64,
+) -> Result<EnclaveRunInfo, String> {
+    let output = Command::new("nitro-cli")
+        .arg("run-enclave")
+        .args(&["--eif-path", image_path])
+        .args(&["--cpu-count", &format!("{}", cpu_count)])
+        .args(&["--memory", &format!("{}", memory_mib)])
+        .output()
+        .map_err(|e| format!("execute nitro-cli error: {}", e))?;
+    parse_output(output)
+}
+
+pub fn run_enclave(opt: &EnclaveOpt, stop_receiver: Receiver<()>) -> Result<(), String> {
+    // check if the enclave already running
+    let enclave_info = describe_enclave()?;
+    if !enclave_info.is_empty() {
+        let info = serde_json::to_string_pretty(&enclave_info).expect("get invalid enclave info");
+        return Err(format!(
+            "the following enclave is already active, please stop and try again:\n{:?}",
+            info
+        ));
+    }
+    // lauch enclave server
+    tracing::info!("start enclave log server at port {}", opt.log_server_port);
+    let enclave_log_server = LogServer::new(
+        opt.log_server_port,
+        opt.log_to_console,
+        opt.log_file.clone(),
+    )
+    .map_err(|e| format!("{:?}", e))?;
+
+    enclave_log_server.launch();
+    // run enclave
+    let info = run_enclave_daemon(&opt.eif_path, opt.cpu_count, opt.memory_mib)?;
+    let s = serde_json::to_string_pretty(&info).unwrap();
+    tracing::info!("run enclave success:\n{}", s);
+    // waiting for stop signal and stop the enclave
+    let _ = stop_receiver.recv();
+    let _ = stop_enclave(Some(info.enclave_id));
+    Ok(())
+}
+
+pub fn stop_enclave(cid: Option<String>) -> Result<EnclaveTerminateInfo, String> {
+    let mut cmd = Command::new("nitro-cli");
+    cmd.arg("terminate-enclave");
+    if let Some(id) = cid {
+        cmd.args(&["--enclave-id", &id]);
+    } else {
+        cmd.arg("--all");
+    }
+    let output = cmd
+        .output()
+        .map_err(|e| format!("execute nitro-cli error: {:?}", e))?;
+    parse_output(output)
+}
+
+pub fn describe_enclave() -> Result<Vec<EnclaveDescribeInfo>, String> {
+    let output = Command::new("nitro-cli")
+        .arg("describe-enclaves")
+        .output()
+        .map_err(|e| format!("execute nitro-cli error: {:?}", e))?;
+    parse_output(output)
+}
+
+pub fn run_vsock_proxy(opt: &VSockProxyOpt) -> Result<(), String> {
+    tracing::debug!("run vsock proxy with config: {:?}", opt);
+    let _ = Command::new("vsock-proxy")
+        .args(&["--num_workers", &format!("{}", opt.num_workers)])
+        .args(&["--config", &opt.config_file])
+        .arg(opt.local_port.to_string())
+        .arg(&opt.remote_addr)
+        .arg(opt.remote_port.to_string())
+        .output()
+        .map_err(|e| format!("execute nitro-cli error: {:?}", e))?;
+    Ok(())
+}
```

Comment:
two questions or possible suggestions:
1. should this also check if the vsock-proxy isn't already running in the background?
2. what happens to this child process when the main process is ended? Given the `output` is thrown out, maybe returning `.spawn()`, so that one can have `Child` and pass the signal to it when exiting?

Reply:
In my test, there is no output.

Reply:
I'm fixing this...